### PR TITLE
meme-bingo-web: init at 0.2.0

### PR DESCRIPTION
**Pull request by auroraanna**
## Description of changes

Added meme-bingo-web package and service using it's executable.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by auroraanna** [comment]
Oh, I forgot the release notes.

File: pkgs/servers/web-apps/meme-bingo-web/default.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
+{ lib, fetchFromGitea, rustPlatform, makeWrapper }:
+
+rustPlatform.buildRustPackage rec {
+  pname = "meme-bingo-web";
+  version = "0.2.0";
+
+  src = fetchFromGitea {
+    domain = "codeberg.org";
+    owner = "annaaurora";
+    repo = pname;
```
Comment (AndersonTorres) [inline pkgs/servers/web-apps/meme-bingo-web/default.nix:10]:
```suggestion
    repo = "meme-bingo-web";
```

File: pkgs/servers/web-apps/meme-bingo-web/default.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    rev = "v${version}";
+    sha256 = "sha256-6hQra+10TaaQGzwiYfL+WHmGc6f0Hn8Tybd0lA5t0qc=";
+  };
+
+  cargoSha256 = "sha256-/hBymxNAzyfapUL5Whkg4NBLA7Fc8A1npXEa9MXTAz4=";
+
+  nativeBuildInputs = [ makeWrapper ];
+  postInstall = ''
```
Comment (AndersonTorres) [inline pkgs/servers/web-apps/meme-bingo-web/default.nix:18]:
```suggestion
  nativeBuildInputs = [ makeWrapper ];

  postInstall = ''
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: nixos/modules/services/web-apps/meme-bingo-web.nix [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services.meme-bingo-web = {
+      description = "A web app for playing meme bingos.";
+      wantedBy = [ "multi-user.target" ];
+
+      environment = env;
```
[the remark this answers is not in the document]
Comment (auroraanna) [reply nixos/modules/services/web-apps/meme-bingo-web.nix:52]: no, I guess it can just be defined below

[the remark this answers is not in the document]
Comment (auroraanna) [reply nixos/modules/services/web-apps/meme-bingo-web.nix]: did that

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Comment by auroraanna** [comment]
I read through the systemd service options manual to determine all the relevant hardening options but I might've missed some.

**Review by AndersonTorres — APPROVED** [review]

**Pull request merged**